game/scripts/vscripts/components/duels/duels.lua

```diff
@@ -47,6 +47,12 @@ function Duels:CheckDuelStatus (keys)
   if not Duels.currentDuel or Duels.currentDuel == DUEL_IS_STARTING or keys.killed:IsReincarnating() then
     return
   end
+  if keys.killed:IsReincarnating() then
+    keys.killed:SetRespawnsDisabled(false)
+  end
+  Timers:CreateTimer(1, function ()
+    keys.killed:SetRespawnsDisabled(true)
```

Comment:
this will race with the end of a duel and cause infinite death.

last kero is killed
timer is created
end duel handler runs
all heroes have respawn enabled
timer runs, disables respawn